### DISCUSSION TITLE
Security insights dashboard

### DISCUSSION THREAD

**Discussion by dduzgun-security** [community/community#55742] [Code Security]
## Problem encountered
The `Security Overview` page helps a lot however it's a snapshot of `today`.

## Suggestion
We would love to have the capability to have a dashboard for security insights using data that is available in GitHub to have visibility on statistics `over time`. 

Something similar to `Activity overview` in `https://github.com/orgs/ORG_NAME/insights` but for security alerts (dependabot, codeql and secrets)

## Features
- View total open vulnerabilities now and from the previous **X period** (decided by the filter: 1 week, 1 month, 3 months, ...) and get a `%` on the progress
- View newly opened vulnerabilities from this month and from the previous **X period** (decided by the filter: 1 week, 1 month, 3 months, ...) and get a `%` on the progress
- View total fixed vulnerabilities from this month and from the previous **X period** (decided by the filter: 1 week, 1 month, 3 months, ...) and get a `%` on the progress
- View the age of resolution of vulnerabilities from this month and from the previous **X period** (decided by the filter: 1 week, 1 month, 3 months, ...) and get a `%` on the progress
- In a graph, view the amount of open vulnerabilities over time by severity.
- Filters for repositories, period time, severity and scans (dependabot, codeql and secrets)

## Expected visual
<img width="1367" alt="image" src="https://github.com/community/community/assets/59659739/af0b6a66-692f-4ac2-81f0-4f0864f9fcc1">

**Comment by kellyarwine** [comment]
👋 Hi @dduzgun-security! Thank you for your post. As the product manager for security overview, I greatly appreciate your feedback. We have some things in progress that I would love to share with you and get your feedback. Do you have 30 minutes to spare? If so, here's [my calendar](removed). _Please note that I will remove this link in 7 days to prevent any potential spam. Looking forward to our conversation!_

**Reply by dduzgun-security** [reply]
Hi @kellyarwine, thanks for the quick reply.
I booked us some time for Tuesday, May 23, 2023

**Reply by kellyarwine** [reply]
I removed the link then.  Feel free to ping me here if you need it again.

**Reply by dduzgun-security** [reply]
@kellyarwine Love the private beta! It looks awesome! Thanks a lot for putting this together.
As a suggestion, having an option to set SLAs and to notify (email, slack, ...) if the vulnerabilities aren't resolved on time would be really useful.

**Reply by kellyarwine** [reply]
@dduzgun-security I've added this to my customer feedback list. Thank you for the suggestion.

**Comment by dduzgun-security** [comment]
Linking a related discussion topic: https://github.com/orgs/community/discussions/43777
